Src/Particle/AMReX_ParticleUtil.H

```diff
@@ -538,7 +538,7 @@ bool enforcePeriodic (P& p,
             }
             // clamp to avoid precision issues;
             if (p.pos(idim) >= rhi[idim]) {
-                p.pos(idim) = static_cast<ParticleReal>(plo[idim]);
```

Comment:
@atmyers I am guessing the reason for the old approach is `rhi` might be >= `phi`, which would fail the assertion later in this function.   Maybe we should change function computeRoundoffDomain to make sure rhi < phi.  We could test if `phi - tolerance` passes the test.  If so, we don't need to the bisect.  If not, we use that as the upper bound for bisect.

Reply:
I will address this in a follow-on PR, where I'll also handle having both a `roundoff_hi_d` and `roundoff_hi_f`